Library/Homebrew/dev-cmd/audit.rb

```diff
@@ -335,11 +335,17 @@ def audit_formula_name
       openssl@1.1
     ].freeze
 
+    PERMITTED_LICENSE_MISMATCHES = [
```

Comment:
This is already being handled in https://github.com/Homebrew/brew/pull/8201

Reply:
It would be handled, but I don't think that checking that the prefix just matches is the right approach in 8201; instead we should have an explicit list of groups that are acceptable if both the GitHub and formula license are in the same group.

Reply:
Fair. I think we could be stricter still here and specifically remap by making the key what GitHub returns and the value what we permit (in case GitHub ends up doing the right thing in future and we are incorrectly allowing these to be used interchangeably).

Reply:
I think this is a better way than what I have in #8201. #8201 needs to wait for Homebrew/core to be updated so this should move ahead and I'll wait to merge #8201 until this is done.